lib/src/ui/checkout_page.dart

```diff
@@ -94,16 +94,19 @@ class _CheckoutPageState extends State<CheckoutPage> {
   void Function() _createAttemptPaymentFunction(BuildContext context, Future<IntentClientSecret> paymentIntentFuture) {
     return () async {
       showProgressDialog(context);
-      final initialPaymentIntent = await paymentIntentFuture.whenComplete(() => hideProgressDialog(context));
+      final initialPaymentIntent = await paymentIntentFuture;
```

Comment:
You need to ensure the progress dialog is hidden if this await fails with an error, or else it will never be hidden

Reply:
yeap, agree, will change it 👌